File: UnitTest/gsd/demo/Module1/ModuleModule1.cs
﻿
using System;
using Zeze.Net;

namespace demo.Module1
{
    public sealed partial class ModuleModule1 : AbstractModule
    {
        public void Start(demo.App app)
        {
        }

        public void Stop(demo.App app)
        {
        }

        protected override long ProcessProtocol1(Protocol p)
        {
            var protocol = p as Protocol1;
            protocol.Send(protocol.Sender);
            Console.WriteLine(protocol.Argument);
            new Rpc2().Send(protocol.Sender, (Protocol) => 0);
            return Zeze.Transaction.Procedure.Success;
        }

        protected override long ProcessProtocol3(Protocol p)
        {
            return Zeze.Transaction.Procedure.NotImplement;
        }

        protected override long ProcessRpc1Request(Protocol p)
        {
            var rpc = p as Rpc1;
            rpc.SendResult();
            return Zeze.Transaction.Procedure.Success;
        }

        protected override long ProcessProtocolNoProcedure(Protocol p)
        {
            throw new NotImplementedException();
        }

        public Table1 Table1 => _Table1;
        public Table2 Table2 => _Table2;
        public TableImportant TableImportant => _TableImportant;
        public tflush Tflush => _tflush;
    }
}

File: ZezeJava/ZezeJavaTest/src/demo/Module1/ModuleModule1.java
package demo.Module1;

// ZEZE_FILE_CHUNK {{{ IMPORT GEN
// ZEZE_FILE_CHUNK }}} IMPORT GEN

public class ModuleModule1 extends AbstractModule {
    public void Start(demo.App app) {
    }

    public void Stop(demo.App app) {
    }

    @Override
    protected long ProcessProtocol3(Protocol3 p) throws Throwable {
        return Zeze.Transaction.Procedure.NotImplement;
    }

    @Override
    protected long ProcessProtocol4(Protocol4 p) throws Throwable {
        return Zeze.Transaction.Procedure.NotImplement;
    }

    @Override
    protected long ProcessRpc1Request(Rpc1 r) throws Throwable {
        return Zeze.Transaction.Procedure.NotImplement;
    }

    public tflush getTflush() {
        return _tflush;
    }

    public Table1 getTable1() {
		return _Table1;
	}

	public Table2 getTable2() {
		return _Table2;
	}

	public Table3 getTable3() {
		return _Table3;
	}

	public TableImportant getTableImportant() {
		return _TableImportant;
	}

    @Override
    public long ProcessProtocolNoProcedure(ProtocolNoProcedure p) {
        return Zeze.Transaction.Procedure.NotImplement;
    }

	// ZEZE_FILE_CHUNK {{{ GEN MODULE
    public ModuleModule1(demo.App app) {
        super(app);
    }
    // ZEZE_FILE_CHUNK }}} GEN MODULE

}
